docs/maven-plugin/src/main/java/org/keycloak/guides/maven/Context.java

```diff
@@ -31,16 +32,30 @@ public Context(File srcDir) throws IOException {
         for (File f : srcDir.listFiles((dir, f) -> f.endsWith(".adoc") && !f.equals("index.adoc"))) {
             Guide guide = parser.parse(f);
 
-            if (guidePriorities != null && guide != null) {
-                Integer priority = guidePriorities.get(guide.getId());
-                guide.setPriority(priority != null ? priority : Integer.MAX_VALUE);
-            }
-
             if (guide != null) {
+                if (guidePriorities != null) {
+                    Integer priority = guidePriorities.get(guide.getId());
+                    if (priority != null) {
+                        if (guide.getPriority() != Integer.MAX_VALUE) {
+                            throw new RuntimeException("Guide is pinned, but has a priority specified: " + f.getName());
+                        }
+                        guidePriorities.remove(guide.getId());
+                        guide.setPriority(priority);
+                    }
+                }
+
+                if (Objects.equals(guide.getTileVisible(), Boolean.FALSE) && guide.getPriority() == Integer.MAX_VALUE) {
+                    throw new RuntimeException("Invisible tiles should be pinned or have an explicit priority: " + f.getName());
+                }
+
                 guides.add(guide);
             }
         }
 
+        if (guidePriorities != null && !guidePriorities.isEmpty()) {
+            throw new RuntimeException("File pinned-guides contained files that do no longer exist or are mis-spelled: " + guidePriorities.keySet());
```

Comment:
There's a few typos here, maybe: `"File 'pinned-guides' contains files that no longer exist or are misspelled: " + ...`